src/operations/ListMembersCloudFileOperation.cpp

```diff
@@ -80,27 +88,34 @@ void Self::onListFetchErrorOccured(CloudFileRequestId requestId, const QString &
     }
 }
 
-void Self::onContactsFetched(const Contacts &requestedContacts, const Contacts &fetchedContacts)
+void Self::onContactsFetched(const quint64 requestId, MutableContacts fetchedContacts)
 {
-    if (requestedContacts == m_membersContacts) {
-        // Build map for quick search
-        std::map<UserId, ContactHandler> map;
-        for (auto &c : fetchedContacts) {
-            map[c->userId()] = c;
-        }
-
-        // Update contacts in members
-        for (auto &m : m_members) {
-            if (const auto c = map[m->contact()->userId()]) {
-                m->setContact(c);
+    if (m_requestId == requestId) {
+        //
+        //  Update contacts in members.
+        //
+        CloudFileMembers updatedMembers;
+        for (auto &&contact : fetchedContacts) {
+            auto currentMember = m_members.find(contact->userId());
+            if (currentMember != m_members.end()) {
+                updatedMembers.push_back(
+                        std::make_unique<CloudFileMember>(currentMember->second->cloneWithContact(contact)));
+                m_members.erase(currentMember);
             }
         }
 
-        // Emit update and finish
+        std::transform(m_members.begin(), m_members.end(), std::back_inserter(updatedMembers),
+                       [this](auto &&member) { return std::move(member.second); });
+
+        m_members.clear();
+
+        //
+        //  Emit update and finish.
+        //
         ListMembersCloudFileUpdate update;
-        update.parentFolder = m_parentFolder;
-        update.file = m_file;
-        update.members = m_members;
+        update.parentFolder = std::move(m_parentFolder);
```

Comment:
In general case, I don't think that reset of class members is a good move. I can't guarantee that they aren't called after reset.